### DISCUSSION TITLE
LSP not showing in cmp window (`nvim_lsp` listed under "unknown source name")

### DISCUSSION THREAD

**Discussion by timii** [hrsh7th/nvim-cmp#1274] [Q&A] [answered]
Hello everyone 👋 

I currently struggle with showing LSP completions in the nvim-cmp window. When using nvim-cmp in .lua files it correctly shows LSP completions, but when editing .js files, it doesn't show up even though the LSP is attached correctly.

When running `:LspInfo` in .js files:
```lua
 Detected filetype:   javascript
 
 2 client(s) attached to this buffer: 
 
 Client: eslint (id: 1, pid: nil, bufnr: [4])
 	filetypes:       javascript, javascriptreact, javascript.jsx, typescript, typescriptreact, typescript.tsx, vue, svelte, astro
 	autostart:       true
 	root directory:  path-to-directory
 	cmd:             cmd.exe /C vscode-eslint-language-server --stdio
 
 Client: null-ls (id: 2, pid: 5001, bufnr: [4])
 	filetypes:       javascript, javascriptreact, jsonc, html, less, graphql, css, markdown.mdx, json, handlebars, yaml, markdown, typescript, scss, typescriptreact, vue, c, cpp, cs, cuda, java, rust, python
 	autostart:       false
 	root directory:  path-to-directory
 	cmd:             <function>
```
As we can see the `eslint` LSP is attached successfully, but now when running `:CmpStatus` it shows the following:
```lua
# ready source names
- buffer
- path
- luasnip

# unavailable source names
- nvim_lua

# unknown source names
- nvim_lsp
```
I added `nvim_lsp` as a source in the cmp sources in my `cmp.lua` file, so I don't really know why this happens:
```lua
local cmp = require("cmp")
local luasnip = require("luasnip")

-- load existing snippets from friendly-snippets
require("luasnip.loaders.from_vscode").lazy_load()

local check_backspace = function()
    local col = vim.fn.col(".") - 1
    return col == 0 or vim.fn.getline("."):sub(col, col):match("%s")
end

cmp.setup({
    snippet = {
        expand = function(args)
            luasnip.lsp_expand(args.body) -- For `luasnip` users.
        end,
    },

    -- Keymappings for the cmp window
    mapping = cmp.mapping.preset.insert({
        -- mappings (i can share it in comments if needed)
    }),

    formatting = {
        -- formatting config (i can share it in comments if needed)
    },
    sources = {
        { name = "nvim_lsp" },
        { name = "nvim_lua" },
        { name = "luasnip" },
        { name = "buffer" },
        { name = "path" },
    },
    confirm_opts = {
        behavior = cmp.ConfirmBehavior.Replace,
        select = false,
    },
})

-- `/` cmdline setup.
cmp.setup.cmdline("/", {
    mapping = cmp.mapping.preset.cmdline(),
    sources = {
        { name = "buffer" },
    },
})

-- `:` cmdline setup.
cmp.setup.cmdline(":", {
    mapping = cmp.mapping.preset.cmdline(),
    sources = cmp.config.sources({
        { name = "path" },
    }, {
        { name = "cmdline" },
    }),
})
```
My neovim version:
```bash
NVIM v0.8.0-1210-gd367ed9b2
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Übersetzt von runneradmin@fv-az177-603

Features: -acl +iconv +tui
```

Is this a problem with nvim-cmp or more of an issue with LSP? 

Any help or idea is greatly appreciated and if you need any further information or parts of my config I will gladly add it in a comment.

Thanks

**Comment by timii** [comment, marked as answer]
After trying out a bit more, I gave up on using `eslint` and tried `tsserver` as an LSP. For whatever reason, it just works now with `tsserver`. So if someone else had the same problem, try `tsserver` if you haven't yet, it will maybe solve your problem, like it did for me. 

If someone stumbles on this question and knows an answer to why `eslint` wasn't working, you can also add it as a comment to help others having the same problem.

**Comment by fitrh** [comment]
This is expected since `eslint` does not provide completion capability,
you can prove it by putting something like this in your `on_attach`

```lua
on_attach = function(client, bufnr)
    if client.name == "eslint" then
        vim.pretty_print(client.server_capabilities)
    end
end
```

An LSP implementation that can provide completions list should show
`completionProvider` as their capability

**Reply by timii** [reply]
Thanks for your response, very good to know. Can I see the server capabilities somewhere external without needing to print it directly within neovim? I couldn't find any listed information about the capabilities on the `eslint` GitHub page or the `mason`/`mason-lspconfig` GitHub page.

**Reply by fitrh** [reply]
You can read them here: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#capabilities

Each LSP implementation should have their capabilities listed somewhere in their repo, but it looks eslint doesn't

**Reply by timii** [reply]
Thank you very much for the help.
